Source/ASCollectionView.mm

```diff
@@ -828,8 +831,12 @@ - (void)performBatchAnimated:(BOOL)animated updates:(void (^)())updates completi
 {
   ASDisplayNodeAssertMainThread();
   [self beginUpdates];
-  if (updates) {
-    updates();
+  as_activity_scope(_changeSet.rootActivity);
```

Comment:
Maybe worth a comment - I don't really understand what this anonymous scope is doing (I think it is only containing the submitActivity, but why is the rootActivity started right above the anonymous scope rather than at the top of the method?)